includes/Modules/AdSense/AMP_Tag.php

```diff
@@ -92,4 +105,40 @@ private function amp_content_add_auto_ads( $content ) {
 		);
 	}
 
+	/**
+	 * Set Web Story Ad Slot ID
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param string $ad_slot_id The Ad Slot ID.
+	 */
+	public function set_story_ad_slot_id( $ad_slot_id ) {
+		$this->story_ad_slot_id = $ad_slot_id;
+	}
+
+	/**
+	 * Adds the AMP Web Story auto ads code if enabled.
+	 *
+	 * @since n.e.x.t
+	 */
+	private function render_story_auto_ads() {
+		printf(
+			'
+				<amp-story-auto-ads>
+					<script type="application/json">
+						{
+							"ad-attributes":
+							{
+								"type": "adsense",
+								"data-ad-client": "%s",
+								"data-ad-slot": "%s"
+							}
+						}
+					</script>
+				</amp-story-auto-ads>
+			',
+			esc_js( $this->tag_id ),
+			esc_js( $this->story_ad_slot_id )
+		);
```

Comment:
Overall, I think it might be a bit more PHP-ish to control this data using a PHP associative array, similar to how we do it for other tags (e.g. `Google\Site_Kit\Modules\Analytics\AMP_Tag`).

How about `printf( '<amp-story-auto-ads><script type="application/json">%s</script></amp-story-auto-ads>', $config )` instead, where `$config` is the PHP array run through `wp_json_encode`?